finagle-mysql/src/main/scala/com/twitter/finagle/mysql/Row.scala

```diff
@@ -71,14 +71,15 @@ class StringEncodedRow(rawRow: Buf, val fields: IndexedSeq[Field], indexMap: Map
       else {
         val str = new String(bytes, Charset(charset))
         field.fieldType match {
-          case Type.Tiny       => ByteValue(str.toByte)
-          case Type.Short      => ShortValue(str.toShort)
-          case Type.Int24      => IntValue(str.toInt)
-          case Type.Long       => IntValue(str.toInt)
-          case Type.LongLong   => LongValue(str.toLong)
-          case Type.Float      => FloatValue(str.toFloat)
-          case Type.Double     => DoubleValue(str.toDouble)
-          case Type.Year       => ShortValue(str.toShort)
+          case Type.Tiny                        => ByteValue(str.toByte)
+          case Type.Short                       => ShortValue(str.toShort)
+          case Type.Int24                       => IntValue(str.toInt)
+          case Type.Long if field.isUnsigned()  => LongValue(str.toLong)
```

Comment:
I didn't get. Longs bigger than 2^63 has Type.LongLong instead of Type.Long.  Should we use isUnsigned check for Type.LongLongs too?